src/community/community.dtos.ts

```diff
@@ -0,0 +1,252 @@
+import {
+    IsOptional,
+    IsArray,
+    IsString,
+    MaxLength,
+    IsNumberString,
+    IsMongoId,
+    IsNotEmpty,
+    ValidateNested, IsIn, IsEthereumAddress, ArrayNotEmpty
+} from 'class-validator';
+import { ApiProperty, OmitType, PartialType, IntersectionType } from '@nestjs/swagger';
+import { Type } from 'class-transformer';
+import { ObjectId } from 'mongodb';
+import constants from '../common/constants';
+
+
+export abstract class CommunityAdministratorBase {
+    @IsNotEmpty()
+    @IsIn(Array.from(constants.PROVIDERS.keys()))
+    @ApiProperty({
+        description: 'Provider id',
+        required: true,
+    })
+        provider_id: string;
+}
+
+export class CommunityAdministratorDiscordDto extends CommunityAdministratorBase {
+
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord user IDs of the administrator',
+        required: false,
+        isArray: true,
+    })
+        user_allowlist: string[];
+    
+    @IsOptional()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsNumberString({},{each:true})
+    @ApiProperty({
+        description: 'Discord role IDs of the administrator',
+        required: false,
+        isArray: true,
+    })
+        role_allowlist: string[];
+}
+
+export class CommunityAdministratorEthereumDto extends CommunityAdministratorBase {
+
+    @IsNotEmpty()
+    @IsArray()
+    @ArrayNotEmpty()
+    @IsEthereumAddress({ each: true })
+    @ApiProperty({
+        description: 'Ethereum addesses of the administrator',
+        required: true,
+        isArray: true,
+    })
+        address_allowlist: string[];
+}
+
+export abstract class CommunityClientConfigBase extends CommunityAdministratorBase {
+}
+
+export class CommunityClientConfigDiscordDto extends CommunityAdministratorBase {
+
+    @IsNotEmpty()
+    @IsNumberString()
+    @ApiProperty({
+        description: 'Discord server This config applies to',
+        required: true,
+    })
+        guild_id: string;
```

Comment:
I wonder if this makes sense as an array in the long term; maybe a community spans multiple servers? Probably not.

Reply:
Yes I thought about that too and I think it would actually be useful, like in bankless dao you could have all the project servers included in the community. I figured it woukd be quite hard to manage if you have a list of multiple servers and then one list with channels from multiple servers etc. So the whole community_config is a list where you can have multiple CommunityClientConfigDiscord's in there.